packages/blocks/src/store/process-block-type.js

```diff
@@ -62,6 +66,24 @@ function mergeBlockVariations(
 	return result;
 }
 
+function stabilizeSupports( rawSupports ) {
```

Comment:
The `stabilizeSupports` function in Gutenberg expects `supports` to be part of a larger block configuration object that doesn’t have getter-only behavior. If `supports` is exported as a standalone object - for example with getter-only behavior - `stabilizeSupports` won’t function as expected.

For example, in Jetpack's case, four blocks were exporting `supports` in a way that led to console errors like `"TypeError: Cannot set property `supports` of #<Object> which has only a getter,"` causing editor load issues (both the site editor and post / page editors will not load at all, if the Jetpack blocks setting is enabled).

We can fix this export approach on our end, but it'd only work for anyone with the version of Jetpack with the fix. I’m also concerned this could affect other plugins as well. Would it be possible to add a conditional check to help prevent similar issues with other plugins using exports like this (and previous versions of plugins)?

Reply:
Thanks for flagging this @coder-karen, this is just the kind of feedback we were hoping for by landing this change in the Gutenberg plugin, so we can account for any particular cases in plugins before this change makes it into a major WP release.

> Would it be possible to add a conditional check to help prevent similar issues with other plugins using exports like this (and previous versions of plugins)?

Can you give me an example of the kind of conditional check you have in mind? How might we mutate the value in a way that's compatible with what's happening in the plugin? Or, do you mean that we would skip any mutations altogether if we only have a getter?

From looking over the code, as far as I can tell, I think the culprit might be here: https://github.com/WordPress/gutenberg/pull/63401#discussion_r1833475175 — where we're mutating `deprecation.supports` instead of creating a new object, which we do slightly further down. So I'm wondering if a fix could be creating a new object with the updated `supports` key instead of mutating `deprecation` directly.

FYI: @aaronrobertshaw for visibility.

Reply:
> I'll look into a fix, but just a heads-up since we've been chatting about the block supports stabilization lately

Happy to test later. It would be good to have a local test condition that reproduces the recursion. 

Is this the relevant Jetpack PR?

- https://github.com/Automattic/jetpack/pull/40070

Reply:
> Happy to test later. It would be good to have a local test condition that reproduces the recursion.

Indeed! Yes, let's see if we can fix while also providing a test to cover this case.

> Is this the relevant Jetpack PR?

That seems to be the one as far as I can tell. Hopefully if we fix this in GB that PR won't be required 🤞

Reply:
I've put up a potential fix over in https://github.com/WordPress/gutenberg/pull/66849

Reply:
Your solution looks ideal here, and works well for Jetpack, thank you!